tools/gen_dir_structure.py

```diff
@@ -0,0 +1,311 @@
+#! /usr/bin/env python3
+
+"""
+Automate the generation of a directory structure
+"""
+
+from argparse import (
+    ArgumentParser,
+    Namespace
+)
+from datetime import (
+    datetime,
+    timedelta
+)
+from pathlib import Path
+from typing import List
+
+import os
+import sys
+
+from stem.descriptor.networkstatus import KeyCertificate, NetworkStatusDocumentV3
+
+from gen_fresh_dirinfo import (
+    consensus_validate_signatures,
+    fetch_certificates,
+    fetch_latest_consensus,
+    generate_certificate,
+    generate_churninfo,
+    generate_customized_consensus,
+    InvalidConsensus,
+    LOGGER
+)
+
+DIRNAME_AUTHORITY_PRIVATE = "private"
+DIRNAME_AUTHORITY_PUBLIC = "public"
+
+FILENAME_AUTHORITY_KEY_ID = "authority_identity_key"
+FILENAME_AUTHORITY_KEY_SIGNATURE = "authority_signing_key"
+FILENAME_AUTHORITY_FINGERPRINT = "authority.txt"
+FILENAME_CERTIFICATE = "certificate.txt"
+FILENAME_CHURN = "churn.txt"
+FILENAME_CONSENSUS = "consensus.txt"
+FILENAME_MICRODESCRIPTORS = "microdescriptors.txt"
+
+# Feel free to modify these values as you see fit.
+CERTIFICATE_VALIDITY_MONTHS = 12
+CONSENSUS_NUMBER_ROUTERS = 120
+CONSENSUS_VALIDITY_DAYS = 7
+
+# Feel free to modify these values as you see fit.
```

Comment:
My comment might be misleading here, because these values never appear to the user, nor are used by Lightarti-rest for building circuits in normal circumstances, but we still need some values to setup the authority and to fill the fields in the consensus.